### PR TITLE
issue-1896/import-message-error-on-eight-rows

### LINKED ISSUE #1896: Import: Message that lists row numbers is wrong for 8 rows
## Description
When there is an error somewhere in the data that a user tries to import, the interface will tell the user what rows contain that error. The interface won't list all rows, but just the first X (is it 8?) rows, and then say "and Y additional rows". But in at least one case, with 8 problems, I got a message saying "and 0 additional rows".

## Steps to reproduce
1. Go to https://app.dev.zetkin.org/organize/1/people
2. Click on "Add..."
3. Click on "Import people"
4. Upload a file that contains more than 10 rows, 8 of which contain malformed email addresses
5. Map columns for first name, last name and email address
6. Click "Validate"

### Expected Behaviour
The message should say something along the lines of "This problem exists on rows 1, 2, 3, 4, 5, 6, 7, 8".

### Actual Behaviour
The message says "This problem exists on rows 1, 2, 3, 4, 5, 6, 7, 8 **and 0 additional rows**"

## Screenshots (if you have any)
image

### PR DESCRIPTION
**Pull request by kaulfield23**
## Description
This PR fixes a bug where an error message displays 'and 0 additional rows' at the end of the sentence if there are more than eight errored rows

## Screenshots
- 7 errored rows
image
- 8 errored rows
image
- 9 errored rows
image
- 10 errored rows
image

## Changes

* Changes `<` to `<=` in `ProblemRowsText`
* Changes messages for singular/plural case

## Related issues
Resolves #1896

### CODE REVIEW

**Review by ziggabyte — CHANGES_REQUESTED** [review]
A nice simple solution! 

However, looking at the screenshots in the PR I saw an error in the messages - here it says "1 additional rows", which should be "1 additional row".

bild

so I looked at the messages and sure enough, that particular message does not have option for single and plural (line 245 in the import messageIds). Even though this is not actually part of the bug, I think it makes sense to fix this when you are in this code and improving this message :) 

So - line 245 needs a to support both single and plural syntax!

**Comment by kaulfield23** [comment]
Oh yeah! singular and plural!!! thank you for spotting this!

File: src/features/import/l10n/messageIds.ts [hunk 1]
```diff
@@ -242,7 +242,7 @@ export default makeMessages('feat.import', {
           'This problem exists on rows {commaRows} and {lastRow}.'
         ),
         manyRows: m<{ additionalRows: number; commaRows: string }>(
-          'This problem exists on rows {commaRows} and {additionalRows} additional rows.'
+          'This problem exists on rows {commaRows} and {additionalRows, plural, =1 {1 additional row} other {# additional rows}}.'
```
Comment (ziggabyte) [inline src/features/import/l10n/messageIds.ts:245]: woop woop!

**Comment by richardolsson** [comment]
Hey comrades I'm sorry for coming back after the fact with this, but I think the reason there was no plural is that this message should actually only be used if there are more than one additional rows. If there is only one more, it would probably be shorter (and more helpful) to just print that number.

Does this make sense? I'm thinking we should probably create another issue for this since this PR has already been merged.

File: src/features/import/l10n/messageIds.ts (hunk 1 above)
Comment (kaulfield23) [reply src/features/import/l10n/messageIds.ts:245]: Yes, it makes sense so that it shows which row has error. I will create an issue for it tomorrow!

**Pull request merged**